exec/cnex/array.c

```diff
@@ -50,17 +62,21 @@ Array *array_resizeArray(Array *self, size_t newSize)
             cell_clearCell(&self->data[i]);
         }
     }
-    if (newSize > 0) {
-        self->data = realloc(self->data, sizeof(Cell) * newSize);
+    if (newSize > 0 && newSize > self->max) {
+        self->max = newSize * 2;
```

Comment:
Again, this seems excessive. If you create an empty array, and then resize it to 1000 elements, you don't want to actually allocate 2000. Instead, you can do 2*max if that's sufficient, or if the requested size is more than that, just allocate the requested size.